pkg/transform/kubernetes/populators.go

```diff
@@ -0,0 +1,158 @@
+		return c, fmt.Errorf("cannot define field 'health' and " +
+			"'livnessProbe' or 'readinessProbe' together")
+	}
+	if c.Health != nil {
+		c.LivenessProbe = c.Health
+		c.ReadinessProbe = c.Health
+		c.Health = nil
+	}
+	return c, nil
+}
+
+func searchConfigMap(cms []spec.ConfigMapMod, name string) (spec.ConfigMapMod, error) {
+	for _, cm := range cms {
+		if cm.Name == name {
+			return cm, nil
+		}
+	}
+	return spec.ConfigMapMod{}, fmt.Errorf("configMap %q not found", name)
+}
+
+func getSecretDataKeys(secrets []spec.SecretMod, name string) ([]string, error) {
+	var dataKeys []string
+	for _, secret := range secrets {
+		if secret.Name == name {
+			for dk := range secret.Data {
+				dataKeys = append(dataKeys, dk)
+			}
+			for sdk := range secret.StringData {
+				dataKeys = append(dataKeys, sdk)
+			}
+			return dataKeys, nil
+		}
+	}
+	return nil, fmt.Errorf("secret %q not found", name)
+}
+
+func convertMapToList(m map[string]string) []string {
+	var d []string
+	for k := range m {
+		d = append(d, k)
+	}
+	return d
+}
+
+func convertEnvFromToEnvs(envFrom []api_v1.EnvFromSource, cms []spec.ConfigMapMod, secrets []spec.SecretMod) ([]api_v1.EnvVar, error) {
+	var envs []api_v1.EnvVar
+
+	// we will iterate on all envFroms
+	for ei, e := range envFrom {
+		if e.ConfigMapRef != nil {
+
+			cmName := e.ConfigMapRef.Name
+
+			// see if the configMap name which is given actually exists
+			cm, err := searchConfigMap(cms, cmName)
+			if err != nil {
+				return nil, errors.Wrapf(err, "envFrom[%d].configMapRef.name", ei)
+			}
+			// once that configMap is found extract all data from it and create a env out of it
+			configMapKeys := convertMapToList(cm.Data)
+			sort.Strings(configMapKeys)
+			for _, key := range configMapKeys {
+				envs = append(envs, api_v1.EnvVar{
+					Name: key,
+					ValueFrom: &api_v1.EnvVarSource{
+						ConfigMapKeyRef: &api_v1.ConfigMapKeySelector{
+							LocalObjectReference: api_v1.LocalObjectReference{
+								Name: cmName,
+							},
+							Key: key,
+						},
+					},
+				})
+			}
+		}
+
+		if e.SecretRef != nil {
+			rootSecretDataKeys, err := getSecretDataKeys(secrets, e.SecretRef.Name)
+			if err != nil {
+				return nil, errors.Wrapf(err, "envFrom[%d].secretRef.name", ei)
+			}
+
+			sort.Strings(rootSecretDataKeys)
+			for _, secretDataKey := range rootSecretDataKeys {
+				envs = append(envs, api_v1.EnvVar{
+					Name: secretDataKey,
+					ValueFrom: &api_v1.EnvVarSource{
+						SecretKeyRef: &api_v1.SecretKeySelector{
+							LocalObjectReference: api_v1.LocalObjectReference{
+								Name: e.SecretRef.Name,
+							},
+							Key: secretDataKey,
+						},
+					},
+				})
+			}
+		}
+	}
+	return envs, nil
+}
+
+func populateEnvFrom(c spec.Container, cms []spec.ConfigMapMod, secrets []spec.SecretMod) (spec.Container, error) {
+	// now do the env from
+	envs, err := convertEnvFromToEnvs(c.EnvFrom, cms, secrets)
+	if err != nil {
+		return c, err
+	}
+	c.EnvFrom = nil
```

Comment:
can you add a comment here about why are we setting it to nil?
The previous code had the following -
```
 -		// Since we are not supporting envFrom in our generated Kubernetes
 -		// artifacts right now, we need to set it as nil for every container.
 -		// This makes sure that Kubernetes artifacts do not container an
 -		// envFrom field.
 -		// This is safe to set since all of the data from envFrom has been
 -		// extracted till this point.
```